docs/deprecations.md

```diff
@@ -23,3 +23,4 @@ being deprecated.
 | [`tekton.dev/task` label on ClusterTasks](https://github.com/tektoncd/pipeline/issues/2533) | [v0.12.0](https://github.com/tektoncd/pipeline/releases/tag/v0.12.0) | Beta | January 30 2021 |
 | [Step `$HOME` env var defaults to `/tekton/home`](https://github.com/tektoncd/pipeline/issues/2013) | [v0.11.0-rc1](https://github.com/tektoncd/pipeline/releases/tag/v0.11.0-rc1) | Beta | December 4 2020 |
 | [Step `workingDir` defaults to `/workspace`](https://github.com/tektoncd/pipeline/issues/1836) | [v0.11.0-rc1](https://github.com/tektoncd/pipeline/releases/tag/v0.11.0-rc1) | Beta | December 4 2020 |
+| [The `TaskRun.Status.ResourceResults.ResourceRef` field is deprecated and will be removed.](https://github.com/tektoncd/pipeline/issues/2694) | [v0.13.0](https://github.com/tektoncd/pipeline/releases/tag/v0.13.0) | Beta | December 4 2020 |
```

Comment:
* maybe we can point them at the field they should use instead as well
* since 0.13.0 is out, we could tentatively put 0.14.0 as the release with the warning
* with our policy [as it stands for beta (assuming this counts as beta? kind of a grey area, im open to treating this as alpha, but it is in the status field of the Task so safer to treat it like beta...)](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#alpha-beta-and-ga) you need at least 9 months warning, so if we do 0.14.0 in july, 9 months would be .... Apr 2021?

Reply:
scope creep, feel free to ignore me: you mentioned issue number 2694 in all the deprecation comments below, it'd probably be useful to have another column in this table to point folks at issues they can reference to track the progress of coming deprecations

Reply:
would it make sense to also mention here that this is also making a dir at the legacy location?

Reply:
Sorry so just to be clear on the policy - there are two changes we would be making.

- Removing the field from our API. The policy is a bit unclear - if we release a v1beta2 without this field, do we need to wait 9 months? https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md
- Writing directories to both locations. This one is more clearly alpha, since it's the behavior of Resources, not the API. I think we can remove this in one more release.

Reply:
> Writing directories to both locations. This one is more clearly alpha, since it's the behavior of Resources, not the API. I think we can remove this in one more release.

Ah, right, that makes sense - and I think we discussed that already, sorry for forgetting!

> Removing the field from our API. The policy is a bit unclear - if we release a v1beta2 without this field, do we need to wait 9 months? https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md

Interesting point - I always assumed this meant it would be 9 months at least until any kind of release went out which make the change in a backwards incompatible way, but I guess once v1beta2 is out, folks could keep using the field via v1beta1 presumably for long enough to meet the 9 month criteria.

I opened https://github.com/tektoncd/pipeline/issues/2790 to discuss. in the meantime i guess just use your best judgement in this PR :D (since there is another entry that says "January 30 2021" i do think other ppl have been assuming 9 months from when the change is introduced)